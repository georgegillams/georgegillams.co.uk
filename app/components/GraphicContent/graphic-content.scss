@import 'helpers/tokens';

.graphic-content {
  &__outer-container {
    position: relative;
    min-width: 15rem;
    height: auto;
    min-height: 10rem;
    justify-content: center;
    align-items: center;
  }

  &__content-container {
    width: 100%;
    transition: all 0.4s ease-in-out;

    &--hidden {
      opacity: 0.2;
      filter: blur(10px);
    }
  }

  &__warning-container {
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;
    background-color: #1e1e1e;

    &__inner {
      display: flex;
      width: 18rem;
      height: 8rem;
      flex-direction: column;
      justify-content: space-around;
      align-items: center;
      color: white;
      text-align: center;
    }
  }

  &__text {
    color: white;
  }
}
