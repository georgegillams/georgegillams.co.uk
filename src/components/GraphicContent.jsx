import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BpkCheckbox from 'bpk-component-checkbox';
import Button from './Button';

import STYLES from './graphic-content.scss';

const getClassName = className => STYLES[className] || 'UNKNOWN';

const GraphicContent = props => {
  const {
    className,
    children,
    graphicContentInView,
    alwaysShowGraphicContent,
    onAlwaysShowChanged,
    onClick,
    ...rest
  } = props;

  const classNameFinal = [getClassName('graphic-content__outer-container')];
  if (className) {
    classNameFinal.push(className);
  }

  const contentContainerClassNames = [
    getClassName('graphic-content__content-container'),
  ];
  if (!graphicContentInView) {
    contentContainerClassNames.push([
      getClassName('graphic-content__content-container--hidden'),
    ]);
  }

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      <div className={contentContainerClassNames.join(' ')}>{children}</div>
      {!graphicContentInView && (
        <div className={getClassName('graphic-content__warning-container')}>
          <div
            className={getClassName(
              'graphic-content__warning-container__inner',
            )}
          >
            <div
              className={getClassName(
                'graphic-content__warning-container--item',
              )}
            >
              This image may contain graphic content
            </div>
            <br />
            <Button
              onClick={onClick}
              className={getClassName(
                'graphic-content__warning-container--item',
              )}
            >
              Show graphic content
            </Button>
            <br />
            <BpkCheckbox
              className={getClassName(
                'graphic-content__warning-container--item',
              )}
              name="Always show graphic content"
              label="Always show graphic content"
              checked={alwaysShowGraphicContent}
              onChange={onAlwaysShowChanged}
            />
          </div>
        </div>
      )}
    </div>
  );
};

GraphicContent.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
  graphicContentInView: PropTypes.bool,
  alwaysShowGraphicContent: PropTypes.bool,
  onClick: PropTypes.func,
  onAlwaysShowChanged: PropTypes.func,
};

GraphicContent.defaultProps = {
  className: null,
  graphicContentInView: false,
  alwaysShowGraphicContent: false,
  onClick: null,
  onAlwaysShowChanged: null,
};

export default GraphicContent;
