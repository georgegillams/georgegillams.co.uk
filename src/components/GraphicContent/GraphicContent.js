import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'gg-components/Checkbox';
import { cssModules } from 'gg-components/helpers/cssModules';
import Button from 'components/common/Button';

import STYLES from './graphic-content.scss';

const getClassName = cssModules(STYLES);

const GraphicContent = props => {
  const {
    className,
    children,
    graphicContentInView,
    alwaysShowGraphicContent,
    onAlwaysShowChanged,
    onClick,
    ...rest
  } = props;

  const classNameFinal = [getClassName('graphic-content__outer-container')];
  if (className) {
    classNameFinal.push(className);
  }

  const contentContainerClassNames = [getClassName('graphic-content__content-container')];
  if (!graphicContentInView) {
    contentContainerClassNames.push([getClassName('graphic-content__content-container--hidden')]);
  }

  const [isServer, setIsServer] = useState(true);

  useEffect(() => {
    setIsServer(false);
  }, []);

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      <div className={contentContainerClassNames.join(' ')}>{children}</div>
      {!graphicContentInView && (
        <div className={getClassName('graphic-content__warning-container')}>
          <div className={getClassName('graphic-content__warning-container__inner')}>
            <div className={getClassName('graphic-content__text')}>This image contains graphic content</div>
            <br />
            {isServer && (
              <div className={getClassName('graphic-content__text')}>Enable JavaScript to view this content</div>
            )}
            {!isServer && (
              <>
                <Button onClick={onClick} className={getClassName('graphic-content__text')}>
                  Show graphic content
                </Button>
                <br />
                <Checkbox
                  labelClassName={getClassName('graphic-content__text')}
                  name="Always show graphic content"
                  label="Always show graphic content"
                  checked={alwaysShowGraphicContent}
                  onChange={onAlwaysShowChanged}
                />
              </>
            )}
          </div>
        </div>
      )}
    </div>
  );
};

GraphicContent.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
  graphicContentInView: PropTypes.bool,
  alwaysShowGraphicContent: PropTypes.bool,
  onClick: PropTypes.func,
  onAlwaysShowChanged: PropTypes.func,
};

GraphicContent.defaultProps = {
  className: null,
  graphicContentInView: false,
  alwaysShowGraphicContent: false,
  onClick: null,
  onAlwaysShowChanged: null,
};

export default GraphicContent;
