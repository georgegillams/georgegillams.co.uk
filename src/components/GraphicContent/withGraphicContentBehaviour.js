import React, { useState } from 'react';
import PropTypes from 'prop-types';
import wrapDisplayName from 'recompose/wrapDisplayName';

export default function withGraphicContentBehaviour(Component) {
  const WithGraphicContentBehaviour = props => {
    const [graphicContentInView, setGraphicContentInView] = useState(false);

    const onClick = () => {
      setGraphicContentInView(true);
    };

    return <Component graphicContentInView={graphicContentInView} onClick={onClick} {...props} />;
  };
  WithGraphicContentBehaviour.displayName = wrapDisplayName(Component, 'withGraphicContentBehaviour');

  WithGraphicContentBehaviour.propTypes = {
    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
    className: PropTypes.string,
  };

  WithGraphicContentBehaviour.defaultProps = {
    style: null,
    className: null,
  };

  return WithGraphicContentBehaviour;
}
