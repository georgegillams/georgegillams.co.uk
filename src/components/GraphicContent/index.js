import GraphicContent from './GraphicContent';
export default GraphicContent;

export withGraphicContentBehaviour from './withGraphicContentBehaviour';
