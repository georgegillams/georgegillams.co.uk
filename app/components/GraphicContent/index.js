import GraphicContent from './GraphicContent';
import withGraphicContentBehaviour from './withGraphicContentBehaviour';

export default GraphicContent;
export { GraphicContent, withGraphicContentBehaviour };
