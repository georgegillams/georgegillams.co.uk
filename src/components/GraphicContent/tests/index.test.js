import React from 'react';

import { render } from '@testing-library/react';

import GraphicContent from '..';

describe('<GraphicContent />', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <GraphicContent>
        <span>Test</span>
      </GraphicContent>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with contentInView', () => {
    const { container } = render(
      <GraphicContent graphicContentInView>
        <span>Test</span>
      </GraphicContent>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with alwaysShow', () => {
    const { container } = render(
      <GraphicContent alwaysShowGraphicContent>
        <span>Test</span>
      </GraphicContent>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(
      <GraphicContent className="test-custom-className">
        <span>Test</span>
      </GraphicContent>
    );

    expect(container).toMatchSnapshot();
  });
});
