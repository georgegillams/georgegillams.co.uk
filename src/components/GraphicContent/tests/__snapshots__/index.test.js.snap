// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraphicContent /> Should render correctly 1`] = `
<div>
  <div
    class="graphic-content__outer-container"
  >
    <div
      class="graphic-content__content-container graphic-content__content-container--hidden"
    >
      <span>
        Test
      </span>
    </div>
    <div
      class="graphic-content__warning-container"
    >
      <div
        class="graphic-content__warning-container__inner"
      >
        <div
          class="graphic-content__text"
        >
          This image contains graphic content
        </div>
        <br />
        <button
          class="sc-dkzDqf kHuqxu graphic-content__text"
          type="button"
        >
          <span
            class="sc-eCYdqJ ghZAyu"
          >
            Show graphic content
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GraphicContent /> Should render correctly with contentInView 1`] = `
<div>
  <div
    class="graphic-content__outer-container"
  >
    <div
      class="graphic-content__content-container"
    >
      <span>
        Test
      </span>
    </div>
  </div>
</div>
`;

exports[`<GraphicContent /> Should render correctly with custom className 1`] = `
<div>
  <div
    class="graphic-content__outer-container test-custom-className"
  >
    <div
      class="graphic-content__content-container graphic-content__content-container--hidden"
    >
      <span>
        Test
      </span>
    </div>
    <div
      class="graphic-content__warning-container"
    >
      <div
        class="graphic-content__warning-container__inner"
      >
        <div
          class="graphic-content__text"
        >
          This image contains graphic content
        </div>
        <br />
        <button
          class="sc-dkzDqf kHuqxu graphic-content__text"
          type="button"
        >
          <span
            class="sc-eCYdqJ ghZAyu"
          >
            Show graphic content
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
